src/app/core/header/header.component.js

```diff
@@ -0,0 +1,31 @@
+import {Component} from '@angular/core';
+
+import style from './header.component.scss';
+import template from './header.component.html';
+
+const HEADER_COMPONENT_SELECTOR = 'sh-header';
+
+@Component({
+    module: module.id,
+    selector: HEADER_COMPONENT_SELECTOR,
+    styles: [style],
+    template,
+    inputs: ['routes']
+})
+export class HeaderComponent {
+
+    mainRoute;
+
+    ngOnInit() {
+        const routes = this.routes.filter(route => route.main);
+        this.mainRoute = routes.length ? routes[0] : null;
```

Comment:
I guess in this case, if you add check ```routes.length``` maybe it will be better to add to condition ```routes && routes.length```, because previous case not cover case when routes  is undefined